### ISSUE TITLE
Take minutes in markdown

### ISSUE THREAD

**Issue by dguest** [indico/indico#3386]
Markdown seems to be supported in a lot of places in indico, but I can't figure out how to enter minutes in markdown format. I seem to only have the option to use the built-in HTML based editor. Could we add an option to add the minutes as markdown? 

I can get around this using browser extensions (i.e. MarkdownHere) but given the wide support that indico has for markdown it seems like it _might_ be a trivial extension.

**Comment by ThiefMaster** [comment]
It's supported in the backend, but we do not have frontend for it at the moment

**Comment by dguest** [comment]
Thanks for the fast response! As I said, right now I'm writing the notes with Markdown Here --- just entering the text into the standard window and then running the conversion. 

It's a bit clumsy though, because editing as "source" prevents Markdown Here from running (because the window is a plane text window) and editing with the rich editor means that return inserts a paragraph break (there may be other oddities too). There are workarounds of course (shift+return seems to do what I want), but it would be nice if there was a way to turn off the auto formatting in the text window too.

**Comment by ThiefMaster** [comment]
If you use a separate markdown editor, convert the notes to HTML, and paste them in the source view of the rich editor widget in Indico.

If you put markdown as text in the editor, it **will not** be converted to HTML even when we do start supporting markdown for minutes (we store in the database whether it's HTML/Markdown, and currently it's always HTML).

**Comment by pferreir** [comment]
It would be great to have a Markdown-based editor for the minutes. It would have to be something simple enough and comparable to CKEditor in terms of functionality, however.

**Comment by ThiefMaster** [comment]
I'd let the user switch between markdown and HTML (with a profile setting for the default) - for less technical users the WYSIWYG editor is probably nicer, but myself I'd rather write markdown (if only to be able to use backticks to mark monospace/code instead of having to go through the huge style list of ckeditor to select code...)

**Comment by dguest** [comment]
Hi @ThiefMaster, this sounds very nice. 

I frequently take notes with a text editor during meetings (especially if the internet connection is lousy), so having the bullet points and citations that markdown provides would be quite a plus.

**Comment by adavidzh** [comment]
> It would have to be something simple enough and comparable to CKEditor in terms of functionality, however.

Regarding `CKEditor` (assuming that that is what `indico` is using) there is a [`markdown` plugin](https://ckeditor.com/cke4/addon/markdown). I don't know what prevents its use.

**Comment by ThiefMaster** [comment]
> Updated: 09 Aug 2015

That doesn't sound very promising.  :/
Quite a few [open issues](https://github.com/hectorguo/CKEditor-Markdown-Plugin/issues) as well.

And I wonder if this actually keeps markdown or produces HTML in the end (which would not be desirable, we want to store markdown when someone writes markdown).

At least I usually don't want any WYSIWYG when writing markdown but write plain markdown and not have to switch back to WYSIWYG mode before saving (which is needed e.g. when using the HTML source view in regular CKEditor).

**Comment by bpedersen2** [comment]
https://github.com/Ionaru/easy-markdown-editor looks promising as a frontend

**Comment by ThiefMaster** [comment]
The first thing I noticed in their demo is the lack of proper monospace/code block support. Neither 4-space indentation nor triple-backtick fences are correctly highlighted in the editor view... :/

**Comment by bpedersen2** [comment]
Seems to be not much different for the github editor here ( using the preview toggle it is correctly styled in the demo.)

**Comment by ThiefMaster** [comment]
Yeah. But the GitHub editor uses pure plaintext in the editing view - IMHO not having a feature at all is better than having a poorly implemented feature ;)

**Comment by pferreir** [comment]
This one doesn't look bad either: https://github.com/HarryChen0506/react-markdown-editor-lite
